[vector-db-sink] Add docs for Pinecone.io, JDBC and Milvus.io (#75)
LangStream/documentation: configuration-resources/data-storage/milvus.md

              "field-types": [
                 {
                    "name": "filename_and_chunkid",
                    "primary-key": true,
                    "data-type": "Varchar",
                    "max-length": 1024
                 },                
                 {
                    "name": "text",
                    "data-type": "Varchar",
                    "max-length": 65535
                 },
                {
                    "name": "language",
                    "data-type": "Varchar",
                    "max-length": 3
                 },
                 {
                    "name": "vector",
                    "data-type": "FloatVector",
                    "dimension": 1536
                 }
              ]
          }
        - |
          {
             "command": "create-index",
             "collection-name": "docs",
             "database-name": "default",
             "field-name": "vector",
             "index-name": "vector_index",
             "index-type": "AUTOINDEX",
             "metric-type": "L2"
          }
        - |
          {
             "command": "load-collection"
          }
```

> As you can see in the "create-statements" you can configure a number of "commands", that translate to Milvus API calls.

Reviewer: ```suggestion
As you can see in the "create-statements" section above, you can configure a number of "commands" that translate to Milvus API calls.
```

The available commands are:
- create-collection: create a collection using the CreateCollection API

> - create-simple-collection: create a collection using the CreateSimpleCollection API, this doesn't need to explicitly specify the field types, create the index on the vector field and load the collection

Reviewer: ```suggestion
- create-simple-collection: create a collection using the CreateSimpleCollection API. This doesn't need to explicitly specify the field types, just create the index on the vector field and load the collection
```

  - name: "lookup-related-documents"
    type: "query-vector-db"
    configuration:
      datasource: "MilvusDatasource"
      query: |
        {
          "collection-name": "docs",
          "vectors": ?,
          "top-k": 10,
          "output-fields": ["text"]
        }
      fields:
        - "value.question_embeddings"
      output-field: "value.related_documents"
``

As usual you can use the '?' symbol as a placeholder for the fields that you specify in the "fields" section.

### Writing to Milvus

When you use the "vector-db-sink" agent to write to Milvus, you can use the following parameters:

```yaml
  - name: "Write to Milvus"
    type: "vector-db-sink"
    input: chunks-topic
    configuration:
      datasource: "MilvusDatasource"
      collection-name: "docs"
      fields:
        - name: "filename_and_chunkid"
          expression: "fn:concat(value.filename, value.chunk_id)"
        - name: "vector"
          expression: "fn:toListOfFloat(value.embeddings_vector)"
        - name: "language"
          expression: "value.language"
        - name: "text"
          expression: "value.text"
        - name: "num_tokens"
          expression: "value.chunk_num_tokens"
```

> You can set the collection-name to the name of the collection you want to write to.

Reviewer: ```suggestion
Set the collection-name to the name of the collection you want to write to.
```

Then you define the fields in the "fields" list.

> This works very like the ['compute' agent](../../pipeline-agents/data-transform/compute.md), and you can use the same syntax to define the fields.

Reviewer: ```suggestion
This works similarly to the ['compute' agent](../../pipeline-agents/data-transform/compute.md), and you can use the same syntax to define the fields.
```
